staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -747,3 +734,203 @@ func determineRequestedIP(svc *v1.Service, fwdRule *compute.ForwardingRule) stri
+	switch v := rule.(type) {
+	case *compute.ForwardingRule:
+		gaRule := rule.(*compute.ForwardingRule)
+		frc = &forwardingRuleComposite{
+			name:           gaRule.Name,
+			ipAddress:      gaRule.IPAddress,
+			description:    &forwardingRuleDescription{APIVersion: meta.VersionGA},
+			backendService: gaRule.BackendService,
+			ports:          gaRule.Ports,
+			ipProtocol:     gaRule.IPProtocol,
+			lbScheme:       gaRule.LoadBalancingScheme,
+			subnetwork:     gaRule.Subnetwork,
+			network:        gaRule.Network,
+		}
+		if gaRule.Description != "" {
+			err = frc.description.unmarshal(gaRule.Description)
+		}
+		return frc, err
+	case *computebeta.ForwardingRule:
+		betaRule := rule.(*computebeta.ForwardingRule)
+		frc = &forwardingRuleComposite{
+			name:              betaRule.Name,
+			ipAddress:         betaRule.IPAddress,
+			description:       &forwardingRuleDescription{APIVersion: meta.VersionBeta},
+			backendService:    betaRule.BackendService,
+			ports:             betaRule.Ports,
+			ipProtocol:        betaRule.IPProtocol,
+			lbScheme:          betaRule.LoadBalancingScheme,
+			subnetwork:        betaRule.Subnetwork,
+			network:           betaRule.Network,
+			allowGlobalAccess: betaRule.AllowGlobalAccess,
+		}
+		if betaRule.Description != "" {
+			err = frc.description.unmarshal(betaRule.Description)
+		}
+		return frc, err
+	default:
+		return nil, fmt.Errorf("Invalid object type %v to compute ForwardingRuleComposite from", v)
+	}
+}
+
+func (f *forwardingRuleComposite) ToBeta() (*computebeta.ForwardingRule, error) {
+	descStr, err := f.description.marshal()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to compute description for beta forwarding rule %s, err: %v", f.name, err)
+	}
+	return &computebeta.ForwardingRule{
+		Name:                f.name,
+		Description:         descStr,
+		IPAddress:           f.ipAddress,
+		BackendService:      f.backendService,
+		Ports:               f.ports,
+		IPProtocol:          f.ipProtocol,
+		LoadBalancingScheme: f.lbScheme,
+		Subnetwork:          f.subnetwork,
+		Network:             f.network,
+		AllowGlobalAccess:   f.allowGlobalAccess,
+	}, nil
+}
+
+func (f *forwardingRuleComposite) ToGA() (*compute.ForwardingRule, error) {
+	descStr, err := f.description.marshal()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to compute description for beta forwarding rule %s, err: %v", f.name, err)
+	}
+	return &compute.ForwardingRule{
+		Name:                f.name,
+		Description:         descStr,
+		IPAddress:           f.ipAddress,
+		BackendService:      f.backendService,
+		Ports:               f.ports,
+		IPProtocol:          f.ipProtocol,
+		LoadBalancingScheme: f.lbScheme,
+		Subnetwork:          f.subnetwork,
+		Network:             f.network,
+	}, nil
+}
+
+type forwardingRuleDescription struct {
+	ServiceName string       `json:"kubernetes.io/service-name"`
+	APIVersion  meta.Version `json:"apiVersion,omitempty"`
```

Comment:
make this consistent
`json:"kubernetes.io/apiVersion,omitempty"`

Reply:
done